tiledb/sm/c_api/tiledb.cc

```diff
@@ -1708,6 +1708,14 @@ int tiledb_query_set_buffer_var(
   if (sanity_check(ctx) == TILEDB_ERR || sanity_check(ctx, query) == TILEDB_ERR)
     return TILEDB_ERR;
 
+  // On writes, check the provided offsets for validity.
+  if (query->query_->type() == tiledb::sm::QueryType::WRITE &&
+      save_error(
+          ctx,
+          tiledb::sm::Query::check_var_attr_offsets(
```

Comment:
Hmm, I would rather move this inside `Query::set_buffer` for var-sized attributes. I think it would be better to gather all errors inside the core TileDB code, exactly where they might occur.